enclave/sgx/qeidinfo.c

```diff
@@ -8,96 +8,103 @@
 #include <openenclave/internal/raise.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include "../common/sgx/qeidentity.h"
-
-/**
- * Validate and copy buffer to enclave memory.
- */
-static oe_result_t _copy_buffer_to_enclave(
-    uint8_t** dst,
-    size_t* dst_size,
-    const uint8_t* src,
-    size_t src_size)
-{
-    oe_result_t result = OE_FAILURE;
-    if (!src || src_size == 0 || !oe_is_outside_enclave(src, src_size) ||
-        dst == NULL || dst_size == NULL)
-        OE_RAISE(OE_INVALID_PARAMETER);
-
-    *dst = malloc(src_size);
-    if (*dst == NULL)
-        OE_RAISE(OE_OUT_OF_MEMORY);
-
-    OE_CHECK(oe_memcpy_s(*dst, src_size, src, src_size));
-    *dst_size = src_size;
-    result = OE_OK;
-
-done:
-    return result;
-}
+#include "internal_t.h"
 
 /**
  * Call into host to fetch qe identity information.
  */
-oe_result_t oe_get_qe_identity_info(oe_get_qe_identity_info_args_t* args)
+oe_result_t oe_get_qe_identity_info(oe_get_qe_identity_info_args_t* args_out)
 {
     oe_result_t result = OE_FAILURE;
-    size_t host_args_buffer_size = sizeof(*args);
-    uint8_t* host_args_buffer = NULL;
-    oe_get_qe_identity_info_args_t* host_args = NULL;
-    oe_get_qe_identity_info_args_t tmp_args = {0};
+    const size_t QE_ID_INFO_SIZE = OE_PAGE_SIZE;
+    const size_t ISSUER_CHAIN_SIZE = OE_PAGE_SIZE;
+    uint32_t retval;
+    oe_get_qe_identity_info_args_t args;
```

Comment:
Consider initializing the struct at definition.

Reply:
I want to see more evidence that initializing the first member zero-fills the whole structure for ancient compilers. Dave mentioned something about supporting C89?

@dthaler : Do you know whether initializing a structure with a single zero value clears the entire structure on old compilers?